volmdlr/voxelization.py

```diff
@@ -0,0 +1,1552 @@
+        f1 = triangle[2] - triangle[1]
+        f2 = triangle[0] - triangle[2]
+
+        # REGION TEST AXES a00..a22 (CATEGORY 3)
+
+        # Test axis a00
+        a00 = np.array([0, -f0[z], f0[y]])
+        p0 = np.dot(v0, a00)
+        p1 = np.dot(v1, a00)
+        p2 = np.dot(v2, a00)
+        r = box_extents[y] * abs(f0[z]) + box_extents[z] * abs(f0[y])
+        if (max(-max(p0, p1, p2), min(p0, p1, p2))) > r:
+            return False
+
+        # Test axis a01
+        a01 = np.array([0, -f1[z], f1[y]])
+        p0 = np.dot(v0, a01)
+        p1 = np.dot(v1, a01)
+        p2 = np.dot(v2, a01)
+        r = box_extents[y] * abs(f1[z]) + box_extents[z] * abs(f1[y])
+        if (max(-max(p0, p1, p2), min(p0, p1, p2))) > r:
+            return False
+
+        # Test axis a02
+        a02 = np.array([0, -f2[z], f2[y]])
+        p0 = np.dot(v0, a02)
+        p1 = np.dot(v1, a02)
+        p2 = np.dot(v2, a02)
+        r = box_extents[y] * abs(f2[z]) + box_extents[z] * abs(f2[y])
+        if (max(-max(p0, p1, p2), min(p0, p1, p2))) > r:
+            return False
+
+        # Test axis a10
+        a10 = np.array([f0[z], 0, -f0[x]])
+        p0 = np.dot(v0, a10)
+        p1 = np.dot(v1, a10)
+        p2 = np.dot(v2, a10)
+        r = box_extents[x] * abs(f0[z]) + box_extents[z] * abs(f0[x])
+        if (max(-max(p0, p1, p2), min(p0, p1, p2))) > r:
+            return False
+
+        # Test axis a11
+        a11 = np.array([f1[z], 0, -f1[x]])
+        p0 = np.dot(v0, a11)
+        p1 = np.dot(v1, a11)
+        p2 = np.dot(v2, a11)
+        r = box_extents[x] * abs(f1[z]) + box_extents[z] * abs(f1[x])
+        if (max(-max(p0, p1, p2), min(p0, p1, p2))) > r:
+            return False
+
+        # Test axis a12
+        a11 = np.array([f2[z], 0, -f2[x]])
+        p0 = np.dot(v0, a11)
+        p1 = np.dot(v1, a11)
+        p2 = np.dot(v2, a11)
+        r = box_extents[x] * abs(f2[z]) + box_extents[z] * abs(f2[x])
+        if (max(-max(p0, p1, p2), min(p0, p1, p2))) > r:
+            return False
+
+        # Test axis a20
+        a20 = np.array([-f0[y], f0[x], 0])
+        p0 = np.dot(v0, a20)
+        p1 = np.dot(v1, a20)
+        p2 = np.dot(v2, a20)
+        r = box_extents[x] * abs(f0[y]) + box_extents[y] * abs(f0[x])
+        if (max(-max(p0, p1, p2), min(p0, p1, p2))) > r:
+            return False
+
+        # Test axis a21
+        a21 = np.array([-f1[y], f1[x], 0])
+        p0 = np.dot(v0, a21)
+        p1 = np.dot(v1, a21)
+        p2 = np.dot(v2, a21)
+        r = box_extents[x] * abs(f1[y]) + box_extents[y] * abs(f1[x])
+        if (max(-max(p0, p1, p2), min(p0, p1, p2))) > r:
+            return False
+
+        # Test axis a22
+        a22 = np.array([-f2[y], f2[x], 0])
+        p0 = np.dot(v0, a22)
+        p1 = np.dot(v1, a22)
+        p2 = np.dot(v2, a22)
+        r = box_extents[x] * abs(f2[y]) + box_extents[y] * abs(f2[x])
+        if (max(-max(p0, p1, p2), min(p0, p1, p2))) > r:
+            return False
```

Comment:
I see that you the same calculus and verification each time. Do you think it could be refactored?

Reply:
I've seen that you have implemented this code also in c. Do you still need this one?

Reply:
It could be refactored, but for now the code is ported "as is" from another repo.
I will take a look at it in the next days.

Yes it is also implemented in C but for now I let the Python implementation for debug purpose.

Reply:
alright, I See. So I can merge it once it passes drone?

Reply:
I am still improving some points regarding some new operational needs. I am passing the PR to draft and will inform you when fully ready for next release!